packages/cli/migrations/src/migrations/0.0.207/add-mode-to-draft-generators/migration.ts

```diff
@@ -0,0 +1,62 @@
+import { AbsoluteFilePath } from "@fern-api/core-utils";
+import { TaskContext, TASK_FAILURE } from "@fern-api/task-context";
+import { readFile, writeFile } from "fs/promises";
+import YAML from "yaml";
+import { Migration } from "../../../types/Migration";
+import { getAllGeneratorYamlFiles } from "./getAllGeneratorYamlFiles";
+
+export const migration: Migration = {
+    name: "add-mode-to-draft-generators-migration",
+    summary: "Adds mode to draft generators in generators.yml. Mode can either be download-files or artifact.",
+    run: async ({ context }) => {
+        const generatorYamlFiles = await getAllGeneratorYamlFiles(context);
+        if (generatorYamlFiles === TASK_FAILURE) {
+            return;
+        }
+        for (const filepath of generatorYamlFiles) {
+            try {
+                await migrateGeneratorsYml(filepath, context);
+            } catch (error) {
+                context.fail(`Failed to migrate ${filepath}`, error);
+            }
+        }
+    },
+};
+
+async function migrateGeneratorsYml(filepath: AbsoluteFilePath, context: TaskContext): Promise<void> {
+    const contents = await readFile(filepath);
+    const parsedDocument = YAML.parseDocument(contents.toString());
+    const draftGenerators = parsedDocument.get("drafts");
+    if (draftGenerators == null) {
+        return;
+    }
+    if (!YAML.isSeq(draftGenerators)) {
+        context.fail(`draft generators are not a list in ${filepath}`);
+        return;
+    }
+    draftGenerators.items.forEach((draftGenerator) => {
+        if (!YAML.isMap(draftGenerator)) {
+            context.fail(`draft generator is not an object in ${filepath}`);
+            return;
+        }
+        const name = draftGenerator.get("name");
```

Comment:
def not worth changing, but for context I just remembered there's a boolean config somewhere that if it's true it'll return you the `Scalar` node with comments vs just the string

Reply:
ah you're right

Reply:
if i pass in true ill get the node as opposed to the `string`